### PR TITLE
Update z3 to 4.8.10.

### PR DESCRIPTION
**Pull request by chriseth**
Depends on https://github.com/ethereum/solidity/pull/10844
Depends on https://github.com/ethereum/solidity/pull/10845
Depends on the builds in https://launchpad.net/~ethereum/+archive/ubuntu/cpp-build-deps

### CODE REVIEW

**Comment by chriseth** [comment]
I already pushed to the PPA.

**Comment by leonardoalt** [comment]
Maybe also already update the docker images?

**Comment by leonardoalt** [comment]
Actually I can do that in the other PR when updating the tests

**Comment by chriseth** [comment]
I don't know what it means to update the docker images.

**Comment by leonardoalt** [comment]
> I don't know what it means to update the docker images.

@chriseth no worries, did it separately

File: scripts/deps-ppa/static_z3.sh [hunk 1]
```diff
@@ -25,7 +25,7 @@ set -ev
 keyid=70D110489D66E2F6
 email=builds@ethereum.org
 packagename=libz3-static-dev
-version=4.8.9
+version=4.8.10
```
Comment (bshastry) [inline scripts/deps-ppa/static_z3.sh:28]: How do we know if the build from this script succeeded?

Comment (leonardoalt) [reply scripts/deps-ppa/static_z3.sh:28]: By checking that the tests in https://github.com/ethereum/solidity/pull/10845 do use z3 4.8.10 (the cmake log shows that)

**Review by cameel — APPROVED** [review]
This one already went through (the package has been pushed to the PPA) and the change in script itself does not affect CI in any way, right? If so, I think this should be merged right away.

**Pull request merged**